spec/lib/reporter/process_spec.rb

```diff
@@ -24,6 +26,50 @@ def check_for(metric, *args)
         :v8_heap_count, :v8_physical_size, :pid, :rss, :thread_count)
     end
 
+    context "job_exception_stats" do
+      before do
+        Discourse.reset_job_exception_stats!
+      end
+
+      after do
+        Discourse.reset_job_exception_stats!
+      end
+
+      it "can collect job_exception_stats" do
+
+        # see MiniScheduler Manager which reports it like this
+        # https://github.com/discourse/mini_scheduler/blob/2b2c1c56b6e76f51108c2a305775469e24cf2b65/lib/mini_scheduler/manager.rb#L95
+        exception_context = {
+          message: "Running a scheduled job",
+          job: { "class" => Jobs::ReindexSearch }
+        }
+
+        2.times do
+          expect {
+            Discourse.handle_job_exception(StandardError.new, exception_context)
+          }.to raise_error(StandardError)
+        end
+
+        metric = Reporter::Process.new(:web).collect
+        expect(metric.scheduled_job_failures).to eq({
+          { "job" => "Jobs::ReindexSearch" } => 2
+        })
+
+        collector = DiscoursePrometheus::Collector.new
+        collector.process(metric.to_json)
+
+        metric = collector.prometheus_metrics.find { |m| m.name == "scheduled_job_failures" }
+
+        expect(metric.data).to eq({
+          {
+            type: "web",
```

Comment:
The type here doesn't seem right given that job failures will only occur in the Sidekiq process.

Reply:
oh it is web, cause the collector runs in a "web" process, I think it is auto detected internally

Reply:
O this is because we initialised it as `Reporter::Process.new(:web)`. Yup in production it'll be the Sidekiq process.